crates/astria-composer/src/collector.rs

```diff
@@ -0,0 +1,38 @@
+use crate::ds::{RollupChainId, RollupTx, RollupTxExt, StreamingClient};
+use color_eyre::eyre::{self};
+use tokio::sync::mpsc as tokio_mpsc;
+
+pub(crate) struct Collector {
+    digestion_channel: tokio_mpsc::UnboundedSender<RollupTxExt>,
+}
+
+impl Collector {
+    pub(crate) fn new() -> (Self, tokio_mpsc::UnboundedReceiver<RollupTxExt>) {
+        let (stream, sink) = tokio_mpsc::unbounded_channel::<RollupTxExt>();
+        let new_collector = Collector {
+            digestion_channel: stream,
+        };
+
+        (new_collector, sink)
```

Comment:
should we be returning the channel reader from the constructor? i think it would be nicer to have a `Collector.subscribe()` method for this and simply returning the `Collector` from the constructor.

deferring to @SuperFluffy

Reply:
@itamarreif can't subscribe to an unbounded mpsc channel unfortunately, that's for watch or broadcast channels.

You indeed need to pass around the sender half of the channel and clone it where needed.

However, I concur that it would be nicer to pass in the rx part of the channel into the constructor rather than return the sender half from it. Just to stay consistent with your rest of the crate.